packages/itwin/tree-widget/src/test/trees/category-tree/CategoriesTreeComponent.test.tsx

```diff
@@ -27,31 +40,215 @@ describe("<CategoriesTreeComponent />", () => {
+          <CategoriesTreeComponent
+            headerButtons={[spy]}
+          />
+        );
+        await waitFor(() => expect(categoryTreeSpy).to.be.called);
+        categoryTreeSpy.args[0][0].onFilterApplied!({ getNodes: async () => categoryNodes } as unknown as IFilteredPresentationTreeDataProvider, 0);
+        await waitFor(() => expect(spy).to.be.calledWith(sinon.match((props: CategoriesTreeHeaderButtonProps) => (props.filteredCategories !== undefined && props.filteredCategories.length === 2))));
+      });
+
+      it("filteredCategories returns filteredCategories with filteredSubcategories when nodes are PresentationTreeNodeItem", async () => {
+        sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+        sinon.stub(UiFramework, "getIModelConnection").returns(iModel);
+        sinon.stub(categoryVisibilityHandler, "useCategories").returns([]);
+        const categoryTreeSpy = sinon.stub(categoryTree, "CategoryTree").returns(<></>);
+        const spy = sinon.stub().returns(<></>);
+        render(
+          <CategoriesTreeComponent
+            headerButtons={[spy]}
+          />
+        );
+        await waitFor(() => expect(categoryTreeSpy).to.be.called);
+        categoryTreeSpy.args[0][0].onFilterApplied!({ getNodes: async () => categoryNodes } as unknown as IFilteredPresentationTreeDataProvider, 0);
+        await waitFor(() => expect(spy).to.be.calledWith(sinon.match((props: CategoriesTreeHeaderButtonProps) => (
+          props.filteredCategories !== undefined
+          && props.filteredCategories.length === 2
+          && props.filteredCategories[0].subCategoryIds!.length === 2
+          && props.filteredCategories[1].subCategoryIds!.length === 0))
+        ));
+      });
+    });
   });
 
   describe("categories tree header buttons", () => {
 
+    it("renders default tree header buttons", () => {
+      const treewHeaderSpy = sinon.stub(treeHeader, "TreeHeader").returns(<></>);
+      sinon.stub(categoryTree, "CategoryTree").returns(<></>);
+      sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+      sinon.stub(UiFramework, "getIModelConnection").returns(iModel);
+      render(<CategoriesTreeComponent />);
+      expect(treewHeaderSpy).to.be.calledWith(sinon.match((props: TreeHeaderProps) => Children.count(props.children) === 3));
+    });
+
+    it("renders user provided tree header buttons", () => {
+      const treewHeaderSpy = sinon.stub(treeHeader, "TreeHeader").returns(<></>);
+      const spy = sinon.stub().returns(<></>);
+      sinon.stub(categoryTree, "CategoryTree").returns(<></>);
+      sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+      sinon.stub(UiFramework, "getIModelConnection").returns(iModel);
+      render(
+        <CategoriesTreeComponent
+          headerButtons={[spy]}
+        />
+      );
+      expect(treewHeaderSpy).to.be.calledWith(sinon.match((props: TreeHeaderProps) => Children.count(props.children) === 1));
```

Comment:
I think it's worth check if spy was called (button rendered)

Reply:
fixed